### PR TITLE
167190672 editing own email fix

### PR DESCRIPTION
**Pull request by jkorpelin**
# Added
* Possibility for users to edit own email

### CODE REVIEW

File: ote/src/clj/ote/services/users.clj [hunk 1]
```diff
@@ -61,17 +61,19 @@
...
+                              (username-exists? db {:username new-username}))
             language (or (:language form-data) :fi)
-            email-changed? (not= (:email requester) (:email form-data))
+            email-changed? (not= (:email user) new-email)
             email-taken? (and email-changed?
-                           (email-exists? db {:email (:email form-data)}))
-            login-info (first (login/fetch-login-info db {:email (:email requester)}))
+                           (email-exists? db {:email new-email}))
+            login-info (first (login/fetch-login-info db {:email (:email user)}))
             password-incorrect? (if admin?
```
Comment (philosotractor) [inline ote/src/clj/ote/services/users.clj]: A matter of taste, & this was already discussed, in any case making a note:
negations and inversions should be avoided in naming for better clarity.
```suggestion
            password-correct? (if admin?
                                             ( ...code...)
                                             false)
```

File: ote/src/clj/ote/services/users.clj [hunk 2]
```diff
@@ -81,31 +83,32 @@
...
           {:success? false
-           :username-taken (when username-taken? (:username form-data))
-           :email-taken (when email-taken? (:email form-data))
+           :username-taken (when username-taken? new-username)
+           :email-taken (when email-taken? new-email)
            :password-incorrect? password-incorrect?}
 
           ;; Request is valid, do update
-          (let [user (specql/update! db ::user/user
+          (let [_ (specql/update! db ::user/user
```
Comment (philosotractor) [inline ote/src/clj/ote/services/users.clj]: Could the `update!` call be moved outside the let?

Comment (jkorpelin) [reply ote/src/clj/ote/services/users.clj]: Sure if it's what we want to do. This pattern has been used quite a bit though.

File: ote/src/cljs/ote/views/confirm_email.cljs [hunk 3]
```diff
@@ -7,12 +7,13 @@
             [ote.ui.common :as common-ui]))
 
 (defn confirm-success
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/confirm_email.cljs]:
```suggestion
(defn- confirm-success
```

File: ote/src/cljs/ote/views/confirm_email.cljs [hunk 4]
```diff
@@ -7,12 +7,13 @@
...
   [:div
    [notification/notification {:text (tr [:register :confirmation-success])
                                :type :success}]
-   [:div {:style {:margin-top "1rem"}}
-    [common-ui/linkify "/#/login" (tr [:login :login-button])]]])
+   (when (:user app)
+     [:div {:style {:margin-top "1rem"}}
+        [common-ui/linkify "/#/login" (tr [:login :login-button])]])])
 
 (defn confirm-failure
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/confirm_email.cljs]:
```suggestion
(defn- confirm-failure
```

**Comment by philosotractor** [comment]
+ Some more localizations would be nice.

Otherwise thanks, LGTM!

**Pull request merged**